voting/tests/common/scenarios.rs

```diff
@@ -0,0 +1,161 @@
+use vote::common::*;
+use vote::general_meeting::{GeneralMeetingId, TimeStamp};
+
+#[derive(Default)]
+pub struct MockDb {
+    map: HashMap<H256, Vec<u8>>,
+}
+
+impl Service for MockDb {}
+
+impl SubStorageAccess for MockDb {
+    fn get(&self, key: &[u8]) -> Option<Vec<u8>> {
+        self.map.get(&blake256(key)).cloned()
+    }
+
+    fn set(&mut self, key: &[u8], value: Vec<u8>) {
+        self.map.insert(blake256(key), value);
+    }
+
+    fn remove(&mut self, key: &[u8]) {
+        self.map.remove(&blake256(key));
+    }
+
+    fn has(&self, key: &[u8]) -> bool {
+        self.map.get(&blake256(key)).is_some()
+    }
+
+    fn create_checkpoint(&mut self) {
+        unimplemented!()
+    }
+
+    fn discard_checkpoint(&mut self) {
+        unimplemented!()
+    }
+
+    fn revert_to_the_checkpoint(&mut self) {
+        unimplemented!()
+    }
+}
+
+fn tx_create_general_meeting(
+    public: &Public,
+    private: &Private,
+    number_of_agendas: u32,
+    end_time: u64,
+    tallying_time: u64,
+) -> Transaction {
+    let tx = vote::general_meeting::TxCreateGeneralMeeting {
+        number_of_agendas,
+        voting_end_time: TimeStamp {
+            time: end_time,
+        },
+        tallying_time: TimeStamp {
+            time: tallying_time,
+        },
+    };
+    let tx = UserTransaction {
+        network_id: Default::default(),
+        action: tx,
+    };
+    let tx_hash = tx.hash();
+    let tx = SignedTransaction {
+        signature: ckey::sign(&tx_hash.as_bytes(), private),
+        signer_public: *public,
+        tx,
+    };
+    Transaction::new("CreateGeneralMeeting".to_owned(), serde_cbor::to_vec(&tx).unwrap())
+}
+
+pub fn test_create_meeting(ctx: &RwLock<Context>) {
+    for stateful in ctx.write().statefuls.values_mut() {
+        stateful.set_storage(ServiceRef::create_export(Box::new(MockDb::default()) as Box<dyn SubStorageAccess>))
+    }
+
+    let admin: Ed25519KeyPair = Random.generate().unwrap();
+    let fake_admin: Ed25519KeyPair = Random.generate().unwrap();
+
+    let admin_key = admin.public();
+
+    ctx.write()
+        .init_genesises
+        .get_mut("general_meeting")
+        .unwrap()
+        .init_genesis(&serde_cbor::to_vec(&admin_key).unwrap());
+
+    let create_meeting = tx_create_general_meeting(&admin.public(), &admin.private(), 2, 12, 18);
+    let create_meeting_invalid = tx_create_general_meeting(fake_admin.public(), fake_admin.private(), 2, 12, 18);
+
+    let create_meeting_invalid_end_time = tx_create_general_meeting(&admin.public(), &admin.private(), 2, 19, 18);
+    let create_meeting_end_passed = tx_create_general_meeting(&admin.public(), &admin.private(), 2, 0, 18);
+    let create_meeting_tallying_passed = tx_create_general_meeting(&admin.public(), &admin.private(), 2, 0, 0);
+
+    let block_hash: BlockHash = BlockHash::default();
+    let author: Public = Public::default();
+    let mut validators: Vec<Public> = Vec::new();
+    validators.push(Public::default());
```

Comment:
Do we need validators?

If we need it, please change the code like this:
```suggestion
    let validators: Vec<Public> = vec![Public::default()];
```
If not, please change the code like this:
```suggestion
    let validators: Vec<Public> = Vec::new();
```